### PR TITLE
Websocket

### PR DESCRIPTION
**Pull request by boomerangll**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,16 +15,21 @@
   "dependencies": {
     "@types/express": "^4.17.3",
```
Comment (AlonShirion) [inline package.json:16]: I think all @types needs to be in devDependencies

File: src/events-manager/events-manager.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+type callbackType = (...params: any) => void;
+
+class EventsManager {
+  eventsMap = new Map<string, callbackType[]>();
+
+  on = (eventName: string, callback: callbackType): (() => void) => {
+    !this.eventsMap.has(eventName) && this.eventsMap.set(eventName, []);
+    this.eventsMap.get(eventName).push(callback);
+
+    return () => this.cancelEvent(eventName, callback);
```
Comment (yonatanbeery) [inline src/events-manager/events-manager.ts:10]: Add a comment that mentions that this function is never excecuted

File: src/events-manager/events-manager.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+type callbackType = (...params: any) => void;
+
+class EventsManager {
+  eventsMap = new Map<string, callbackType[]>();
+
+  on = (eventName: string, callback: callbackType): (() => void) => {
```
Comment (yonatanbeery) [inline src/events-manager/events-manager.ts:6]: Change function name to something like 'add_event_callback_function'. The name 'on' is not self-explaining

File: src/bitbucket-data-map/butbucket-data-map.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import {BitbucketData} from '../bitbucket-data/bitbucket-data';
```
Comment (yonatanbeery) [inline src/bitbucket-data-map/butbucket-data-map.ts:1]: Change the file name from 'butbucket-data-map' to 'bitbucket-data-map'

File: src/bitbucket-data-map/butbucket-data-map.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (yonatanbeery) [inline src/bitbucket-data-map/butbucket-data-map.ts:2]: Add a note for everyone who tries to read this file

File: src/index.ts [hunk 6]
```diff
@@ -1,70 +1,6 @@
...
-import {Request, Response} from 'express';
-import {
-  forkRepository,
-  getAllProjectNames,
-  getAllSeedRepoTypes,
-  getAllRepositories,
-  dataForBamboo,
-} from './atlasian-utils';
-import * as asyncHandler from 'express-async-handler';
+import {createHttpServer} from './bitbucket-connector-http-server/bitbucket-connector-http-server';
```
Comment (yonatanbeery) [inline src/index.ts:1]: General note: chenge the logs to understandble ones (not 'kakatov').

File: src/bitbucket-data-storage/bitbucket-data-storage.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import BitbucketDataMap, {
+  DataTypes,
+} from '../bitbucket-data-map/butbucket-data-map';
+import EventsManager from '../events-manager/events-manager';
+import {BitbucketData} from '../bitbucket-data/bitbucket-data';
+import {ProjectsData} from '../bitbucket-data/projects-data/projects-data';
+import {SeedRepositoriesData} from '../bitbucket-data/seed-repositories-data/seed-repositories-data';
+
+const INTERVAL_TIME = 10 * 1000;
+const DATA_UPDATE_EVENT = 'dataUpdate';
```
[the remark this answers is not in the document]
Comment (boomerangll) [reply src/bitbucket-data-storage/bitbucket-data-storage.ts:10]: i wont do that because then the code wont be generic

**Pull request merged**